addon/components/frost-expand.js

```diff
@@ -0,0 +1,139 @@
+/**
+ * Component definition for the frost-expand component
+ */
+import Ember from 'ember'
+const {observer} = Ember
+import layout from '../templates/components/frost-expand'
+import computed, {readOnly} from 'ember-computed-decorators'
+import Component from './frost-component'
+import {PropTypes} from 'ember-prop-types'
+
+export default Component.extend({
+  // == Dependencies ==========================================================
+
+  // == Keyword Properties ====================================================
+
+  layout,
+  tagName: 'div',
+  classNameBindings: ['cssStateClass'],
+
+  // == PropTypes =============================================================
+
+  /**
+   * Properties for this component. Options are expected to be (potentially)
+   * passed in to the component. State properties are *not* expected to be
+   * passed in/overwritten.
+   */
+  propTypes: {
+    // options
+    onExpand: PropTypes.func,
+    onCollapse: PropTypes.func,
```

Comment:
Ok, but this isn't an input component. For example, when using it in the error page, do we really care about each time it is expanded or collapsed? We're just providing some content to the user that they can choose whether they want to view or not but we don't care about every time they toggle the visibility of the content. Just my 2 cents, I'm fine with doing it either way